### ISSUE TITLE
Get containerd version 1.7 into Garden Linux

### ISSUE THREAD

**Issue by nanory** [gardenlinux/gardenlinux#1782]
**What would you like to be added**:
As discussed in some of our meetings, we would like to add the containerd version 1.7 to Garden Linux.
Currently, there are no plan to get containerd 1.7 in Debian testing by the maintainer, so it might be up to us to provide this package in our packaging pipeline.

For this, there is already a corresponding packaging repo which currently uses trixie sources for creating the packages out of it: https://gitlab.com/gardenlinux/gardenlinux-package-containerd

We should consider to create a PR there, so that is probably uses the following sources:
https://github.com/containerd/containerd

**Why is this needed**:
Some users requesting to have the new containerd version 1.7 available.

**Comment by alee-ntap** [comment]
Checked that the `containerd` version `1.7.x` isn't a LTS version. And it will be end of life on March 10, 2024 or release of 2.0 + 6 months. Details from:
    https://github.com/containerd/containerd/blob/main/RELEASES.md#support-horizon

**Comment by alee-ntap** [comment]
Debian planed to release version 13 on June 2025 for add RISC-V support. So we expected it will be soon be freeze on 2024. The `containerd` maintainer definitely do not want non-LTS version as soon will be end of life version `1.7.x` into Debian `testing` before next TLS version of `containerd` available.

**Comment by alee-ntap** [comment]
I found ubuntu introduced `containerd` version `1.7.2` and then removed it afterwards. They made `containerd` version `1.6` source package ship only the -dev package directly from Debian. And packaged `containerd-app` version `1.7.2` with vendorized dependency, so that they can build easily on all supported Ubuntu versions. I think it's a very smart way to keep the -dev as is from Debian. So that won't need to be worry about how to build and tests all the packages which reverse depends.

As we still want to use version `1.6` in next major release. So it's better to build package with different for early test. So I created `containerd-1.7` package based on Ubuntu’s `container-app`'s debian/ folder with additional modification to fits our needs. It contains the `patches` from `Ubuntu` with enabled `tests` from `upstream` at build time on `amd64`. I specifically made the binary package name called `containerd-1.7`. So that it won’t override on it’s newer version on exist `containerd` package in our repository, and our users won’t get upgraded accidentally without additionally specific `-1.7` in the package name for installation and image build.
